curp/tests/read_state.rs

```diff
@@ -0,0 +1,34 @@
+use std::time::Duration;
+
+use curp::{client::ReadState, cmd::Command};
+use utils::config::ClientTimeout;
+
+use crate::common::{curp_group::CurpGroup, init_logger, sleep_millis, test_cmd::TestCommand};
+
+mod common;
+
+#[tokio::test]
+async fn read_state() {
+    init_logger();
+    let group = CurpGroup::new(3).await;
+    let put_client = group.new_client(ClientTimeout::default()).await;
+    let put_cmd = TestCommand::new_put(vec![0], 0).set_exe_dur(Duration::from_millis(100));
+    let put_id = put_cmd.id().clone();
+    tokio::spawn(async move {
+        assert_eq!(put_client.propose(put_cmd).await.unwrap(), vec![]);
+    });
+    let get_client = group.new_client(ClientTimeout::default()).await;
+    let res = get_client
+        .fetch_read_state(&TestCommand::new_get(vec![0]))
+        .await
+        .unwrap();
+    assert_eq!(res, ReadState::Ids(vec![put_id]));
```

Comment:
How about modifying the code like this? Doing so, we don't need to derive `PartialEq` and `Eq` for `ReadState`. 
```suggestion
    if let ReadState::Ids(v) = res {
        assert_eq!(v, vec![put_id])
    } else {
        unreachable!("expected result should be ReadState::Ids({:?}, but received {:?}", vec![put_id], res);
    }
    
    assert!(matches!(res, ReadState::CommitIndex(1)));
```

Reply:
Fixed